### PR TITLE
Updated proposal to include RAF and Commit

### PR DESCRIPTION
**Pull request by fserb**
1st draft.

### CODE REVIEW

File: OffscreenCanvasAnimation.md [hunk 1]
```diff
@@ -214,20 +215,38 @@ interface HTMLVideoElementFrameRequestData : FrameRequestData {
...
+  readonly attribute OffscreenCanvas canvas;
+};
+
+interface OffscreenCanvasRenderingContext2D : OffscreenRenderingContext {};
+interface OffscreenWebGLRenderingContext : OffscreenRenderingContext, WebGLRenderingContext {};
+```
+
+#### The commit processing model
+
+When `commit()` is called, a copy of the frame is made and sent to the driver/display when appropriate. This function can return immediately or block. The UA can decide to queue a few frames before blocking to guarantee smoothness.
```
Comment (junov) [inline OffscreenCanvasAnimation.md]: Actually, we don't always make a copy. If it is a WebGL context with preserveDrawingBuffer = false, we transfer the frame instead.

File: OffscreenCanvasAnimation.md [hunk 2]
```diff
@@ -214,20 +215,38 @@ interface HTMLVideoElementFrameRequestData : FrameRequestData {
...
+
+
+#### Rationale
+* RAF allows for animations to be synced and support current rendering models on the web.
+* commit() supports the use case of a continuously running task that produces frames repeatedly over time, wihch enables the port of native application to the Web (e.g. via Emscripten).
```
Comment (junov) [inline OffscreenCanvasAnimation.md]: wihch -> which

Comment (junov) [reply OffscreenCanvasAnimation.md]: I think the context here could be developped a bit more. Should mention WASM, and how following an async model requires exiting and re-entering the WASM scope, which is bad.

File: OffscreenCanvasAnimation.md (hunk 1 above)
Comment (junov) [reply OffscreenCanvasAnimation.md]: Missing: mechanism for throttling the frame rate

**Review by junov — CHANGES_REQUESTED** [review]

**Comment by fserb** [comment]
Done. PTAL.

File: OffscreenCanvasAnimation.md [hunk 3]
```diff
@@ -214,20 +215,38 @@ interface HTMLVideoElementFrameRequestData : FrameRequestData {
...
+  readonly attribute OffscreenCanvas canvas;
+};
+
+interface OffscreenCanvasRenderingContext2D : OffscreenRenderingContext {};
+interface OffscreenWebGLRenderingContext : OffscreenRenderingContext, WebGLRenderingContext {};
+```
+
+#### The commit processing model
+
+When `commit()` is called, the frame is sent to display. This function will block until another frame can be started to render. The UA can decide to queue a few frames before blocking to guarantee smoothness.
```
Comment (junov) [inline OffscreenCanvasAnimation.md]: there is no "guarantee", only hope.

**Review by junov — CHANGES_REQUESTED** [review]

**Review by junov — APPROVED** [review]

**Pull request merged**